src/it/circular-project/invoker.properties

```diff
@@ -0,0 +1,3 @@
+invoker.goals=enforcer:enforce
+invoker.buildResult = failure

```

Comment:
This IT fails indeed, but I'm not sure it happens in expected way:
```
[ERROR] Child module .../target/it/circular-project/slf4j-api of .../target/it/circular-project/pom.xml does not exist @ 
[ERROR] 'packaging' with value 'jar' is invalid. Aggregator projects require 'pom' as packaging. @ line 2, column 106
 @ 
[ERROR] The build could not read 1 project -> [Help 1]
```

Reply:
With changes as suggested for `it/.../pom.xml` - build of `circular-project` completes successfully, (which is test failure, as `banProjectCircularDependencies` is not triggered).

Reply:
Thanks - it seems I've been misled by the local build - you are correct that it seems to be failing for a different reason (the pom-packaging) and when that is corrected there is an error being logged which indicates that the trick of trying to re-use an existing project isn't working:
[WARNING] The POM for org.slf4j:jcl-over-slf4j:jar:1.6.1 is invalid, transitive dependencies (if any) will not be available: 2 problems were encountered while building the effective model for org.slf4j:jcl-over-slf4j:1.6.1
[ERROR] 'dependencies.dependency.version' for org.slf4j:slf4j-api:jar is missing. @ 
[ERROR] 'dependencies.dependency.version' for org.slf4j:slf4j-jdk14:jar is missing. @ 

Seems that its a bit tricky to set up a good test case for this. Will have another look and see if I can set up projects directly for the test.

Reply:
I've added specific version dependencies to the pom for the it and the build now fails correctly because of the rule with message:
[WARNING] Rule 0: org.apache.maven.plugins.enforcer.BanProjectCircularDependencies failed with message:
Circular Project Dependency found. No modules groupId:artifactId combination in your project must exist in the list of transitive dependencies for any module in the project.
  org.slf4j:slf4j-api found in transitive dependencies of org.slf4j:childb

Reply:
> the build now fails correctly because of the rule with message:

Perhaps this could be verified during the build? For example https://github.com/mojohaus/extra-enforcer-rules/blob/52017dd8ff5e9ed7356c4e8f4d52ba070a1ba03f/src/it/mojo-1853/invoker.properties#L4 also is expecting build failure and in https://github.com/mojohaus/extra-enforcer-rules/blob/52017dd8ff5e9ed7356c4e8f4d52ba070a1ba03f/src/it/mojo-1853/verify.groovy#L4 the reason seems to be checked.

Just to make sure that only the specific failure is accepted as success, and not _any_ failure.

Reply:
Great idea, thanks - will add that.